pkg/jobs/jobspb/schedule.proto

```diff
@@ -44,6 +47,19 @@ message ScheduleDetails {
 
   // How to handle failed jobs.
   ErrorHandlingBehavior on_error = 2;
+
+  // ClusterID is populated at schedule creation with the ClusterID, in case a
```

Comment:
I'm going to keep what i have for now, as I don't like that it would be up to the author to maintain the invariant. Depending on how I implement the follow up PR, I may change the comment to your suggestion.